### ISSUE TITLE
Java NPE on CMS Webdav door

### ISSUE THREAD

**Issue by ageorget** [dCache/dcache#7188]
Hi,

We had a GGUS ticket yesterday about CMS SAM tests failing with timeout errors : 

```
PROPFIND of https://ccdavcms.in2p3.fr:2880/disk/data/store/
Failed to send PROPFIND request to ccdavcms.in2p3.fr:2880 within timeout: The read operation timed out
```
https://monit-grafana.cern.ch/goto/l00aUkX4k?orgId=20

I checked in the Webdav door logs and found some NPE errors which matched the time when SAM tests failed : 

```
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: 14 Jun 2023 10:56:32 (webdav-ccdcatli345) [door:webdav-ccdcatli345@webdav-ccdcatli345Domain:AAX+ExmxeAA] Internal server error
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: java.lang.NullPointerException: null
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at java.base/java.util.ArrayList.addAll(ArrayList.java:702)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at io.milton.http.DeleteHelperImpl.isLockedOut(DeleteHelperImpl.java:65)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at io.milton.http.http11.DeleteHandler.processExistingResource(DeleteHandler.java:92)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at io.milton.http.ResourceHandlerHelper.processResource(ResourceHandlerHelper.java:196)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at io.milton.http.ResourceHandlerHelper.processResource(ResourceHandlerHelper.java:131)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at io.milton.http.http11.DeleteHandler.processResource(DeleteHandler.java:83)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at io.milton.http.ResourceHandlerHelper.process(ResourceHandlerHelper.java:127)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.dcache.webdav.DcacheResourceHandlerHelper.process(DcacheResourceHandlerHelper.java:42)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at io.milton.http.http11.DeleteHandler.process(DeleteHandler.java:72)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.dcache.webdav.DcacheStandardFilter.process(DcacheStandardFilter.java:50)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at io.milton.http.FilterChain.process(FilterChain.java:46)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.dcache.webdav.transfer.CopyFilter.process(CopyFilter.java:276)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at io.milton.http.FilterChain.process(FilterChain.java:46)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at io.milton.http.HttpManager.process(HttpManager.java:158)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.dcache.webdav.MiltonHandler.handle(MiltonHandler.java:77)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:59)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.dcache.http.AuthenticationHandler.access$001(AuthenticationHandler.java:55)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.dcache.http.AuthenticationHandler.lambda$handle$0(AuthenticationHandler.java:156)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at java.base/java.security.AccessController.doPrivileged(Native Method)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at java.base/javax.security.auth.Subject.doAs(Subject.java:361)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.dcache.http.AuthenticationHandler.handle(AuthenticationHandler.java:153)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:59)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.dcache.http.AbstractLoggingHandler.handle(AbstractLoggingHandler.java:110)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:59)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.rewrite.handler.RewriteHandler.handle(RewriteHandler.java:322)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.server.Server.handle(Server.java:516)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:400)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:645)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:392)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.onFillable(SslConnection.java:555)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:410)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:164)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:409)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
Jun 14 10:56:32 ccdcatli345 dcache@webdav-ccdcatli345Domain[114454]: at java.base/java.lang.Thread.run(Thread.java:829)
```
Full log and access log available here : https://filesender.renater.fr/?s=download&token=6d740250-5a70-43af-b49c-0f8ffbdd826e
Restarting the Webdav door made the SAM tests green again.

Cheers,
Adrien

**Comment by paulmillar** [comment]
This problem should be fixed by upgrading dCache to either 8.2.18 (or newer) or 9.0.6 (or newer).

**Comment by ageorget** [comment]
Hi Paul,
Thanks for your quick answer. As we're running 8.2.16, I will plan to upgrade soon.
